codemon/codemon.py

```diff
@@ -18,26 +20,47 @@ def main():
       countArg+=1;
 
       if arg == "init":
+        if sys.argv[countArg] == '-cpp':
+          if sys.argv[countArg] == '-n':
+            file = sys.argv[countArg+1]
+            path = '.'
+            f = open(path + '/' + file + '.cpp',"w+")
+            template = template_cpp()
+            f.write(template)
+            f.close()
+            print(colored.yellow("Created "+file+'.cpp'))
+            break;
 
-        if sys.argv[countArg] == '-n':
-          file = sys.argv[countArg+1]
-          path = '.'
-          f = open(path + '/' + file + '.cpp',"w+")
-          template = template_cpp()
-          f.write(template)
-          f.close()
-          print(colored.yellow("Created "+file+'.cpp'))
-          break;
-
-        else:
-          contestName = sys.argv[countArg]
-          fileNames = get_filename()
-          init(contestName, fileNames)
+          else:
+            contestName = sys.argv[countArg]
+            fileNames = get_filename()
+            init(contestName, fileNames)
+
+        elif sys.argv[countArg] == '-java':
+          
+          if sys.argv[countArg] == '-n':
+            file = sys.argv[countArg+1]
+            path = '.'
+            f = open(path + '/' + file + '.java',"w+")
+            template = template_java()
+            f.write(template)
+            f.close()
+            print(colored.yellow("Created "+file+'.java'))
+            break;
+          else:
+            contestName = sys.argv[countArg]
+            fileNames = get_filename_java()
+            initjava(contestName, fileNames)
 
```

Comment:
You may notice that we are repeating the same stuff from Cpp part here. We don't want to duplicate code. Refactor this entire function such that we can have the least amount of code duplication here. This can be written as: 
```python
def main():

  if len(sys.argv) < 2:
    showHelp()

  else:
    countArg = 0;
    isCppFile = False
    isJavaFile = False
    isSingleFile = False
    toInit = False
    toListen = False
    toPractice = False
    targetFile = ""
    # get all arguments and flags
    for arg in sys.argv:
      countArg+=1;
      if arg == "init":
        toInit = True
      else if arg == "-cpp":
        isCppFile = True
      else if arg == "-java":
        isJavaFile = True
      else if arg == "-n":
        isSingleFile = True
      else if arg == "listen":
        isListen = True
      else if arg == "practice":
        toPractice = True
      else:
        targetFile = arg

    # based on booleans perform action
    if toListen == True:
      listen()
    else if toInit == True:
      if isJavaFile == True and isSingleFile == False:
        extension = ".java" # we can also pass it directly in function
        path = "."
        # create that directory with that given name (targetFile)
        createTargetDirectory(path, template, targetFile, extension)
      elif isJavaFile == True and isSingleFile == True:
        # extension & path passed directly here
        createTargetFile('.', template, targetFile, 'java') 
      elif isCppFile == True and isSingleFile == False:
      # and so on 
        .....
```